doc/users/colormapnorms.rst

```diff
@@ -0,0 +1,123 @@
+.. _colormapnorm-tutorial:
+
+Colormap Normaliztions 
+================================
+
+Objects that use colormaps usually linearly map the colors in the
+colormap from data values ``vmin`` to ``vmax``.  For example::
+
+    pcm = ax.pcolormesh(x, y, Z, vmin=-1., vmax=1., cmap='RdBu_r')
+
+will map the data in *Z* linearly from `-1` to `+1`, so *Z=0* will
+give a color at the center of the colormap *RdBu_r* (white in this
+case).  
+
+Normalizations are defined as part of :func:`matplotlib.colors`
+module.  The default normalization is
+:func:`matplotlib.colors.Normalize`.  The artists that map data to
+color pass the arguments ``vmin`` and ``vmax`` to
+:func:`matplotlib.colors.Normalize`. We can
+substnatiate the normalization and see what it returns.  In this case
+it returns 0.5::
+
+    norm = mpl.colors.Normalize(vmin=-1., vmax=1.)
+    norm(0.)
+
+However, there are sometimes cases where it is useful to map data to
+colormaps in a non-linear fashion. 
+
+Logarithmic
+---------------------------------
+
+One of the most common transformations is to plot data by taking its
+logarithm (to the base-10).  This transofrmation is useful when there
+are changes across disparate scales that we still want to be able to
+see.  Using :func:`colors.LogNorm` normalizes the data by :math:`log_{10}`.  In
+the example below, there are two bumps, one much smaller than the
+other. Using :func:`colors.LogNorm`, the shape and location of each
+bump can clearly be seen:
+
+.. plot:: users/plotting/examples/colormap_normalizations_lognorm.py
+   :include-source:
+
+Symetric logarithmic
+---------------------------------
+
+Similarly, it sometimes happens that there is data that is positive
+and negative, but we would still like a logarithmic scaling applied to
+both.  In this case, the negative numbers are also scaled
+logarithmically, and mapped to small numbers.  i.e. If `vmin=-vmax`, then
+they the negative numbers are mapped from 0 to 0.5.  
+
+Since the values close to zero tend toward infinity, there is a need
+to have a range around zero that is linear.  The parameter *linthresh*
+allows the user to specify the size of this range (-*linthresh*,
+*linthresh*).  The size of this range in the colormap is set by
+*linscale*.  When *linscale* == 1.0 (the default), the space used for
+the positive and negative halves of the linear range will be equal to
+one decade in the logarithmic range.
+
+.. plot:: users/plotting/examples/colormap_normalizations_symlognorm.py
+   :include-source:
+
+Power-law
+---------------------------------
+
+Sometimes it is useful to remap the colors onto a power-law
+relationship (i.e. :math:`y=x^{\gamma}`, where :math:`\gamma` is the
+power).  For this we use the :func:`colors.PowerNorm`.  It takes as an
+argument *gamma* ( *gamma* == 1.0 will just yield the defalut linear
+normalization):
+
+.. note::
+  
+   There should probably be a good reason for plotting the data using
```

Comment:
On the other hand, this is also called 'gamma' and is done to almost every digital image at some point in the processing.